### PR TITLE
feat: Inter fonts

### PR DESCRIPTION
**Pull request by kyhyco**
**What changed? Why?**
- Apply `Inter` font to doc site

Example
| Old | New |
| --- | --- |
|  Screenshot 2024-06-10 at 2 25 00 PM | Screenshot 2024-06-13 at 2 39 42 PM |

**Notes to reviewers**
Tried out several approaches:
- importing `@fontsource/inter`: messes up the css import path and will require creating a separate package just to handle fonts if we go this route
- including Inter font files in OnchainKit: this could be an alternative if we don't want to directly hit google fonts
- using `@tailwind/components` to create custom font components: much better to create a react component

- Use case for tailwind and non-tailwind users is supported in this PR

**How has it been tested?**
locally

### CODE REVIEW

File: site/docs/pages/App.tsx [hunk 1]
```diff
@@ -32,7 +32,9 @@ export default function App({ children }: { children: ReactNode }) {
   return (
     <WagmiProvider config={wagmiConfig}>
       <QueryClientProvider client={queryClient}>
-        <div style={{ display: 'flex', flexDirection: 'column' }}>{children}</div>
+        <div style={{ display: 'flex', flexDirection: 'column' }} className="ock-docsite-content">
```
Comment (Zizzamia) [inline site/docs/pages/App.tsx:35]: Do we want all our demo to shocase Inter?

Comment (kyhyco) [reply site/docs/pages/App.tsx:35]: Great question. Let's check in with design tomorrow

**Comment by mindapivessa** [comment]
@kyhyco I don't think the font applied is Inter? 

Here's Inter: https://fonts.google.com/specimen/Inter

Screenshot 2024-06-11 at 12 26 39 PM

File: src/text/TextBody.tsx [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import { ReactNode } from 'react';
+
```
Comment (Zizzamia) [inline src/text/TextBody.tsx:2]: Why `src/text`?

Comment (Zizzamia) [reply src/text/TextBody.tsx:2]: Oh I see, you are trying to create an internal Design System to use for building all those pieces.

got it!

Comment (kyhyco) [reply src/text/TextBody.tsx:2]: We could also do something like `src/internal/text`

**Review by abcrane123 — APPROVED** [review]
love this !

**Review by Zizzamia — APPROVED** [review]

**Pull request merged**